analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -713,5 +713,23 @@ public void EventReference_SetsNotNull()
             validator.ValidateTag("AfterAdd", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
             validator.ValidateTag("AfterRemove", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
         }
+
+        [TestMethod]
+        public void ReDim_SetsNotNull()
+        {
+            const string code = @"
+Dim First(), Second(), Third(4242) As Object
```

Comment:
Add an array declaration with an initializer
```
Dim arr() = New Object() { }
```

Reply:
Add Tag calls, so we can see `First` and `Second` are null but `Third` isn't before the ReDim.

Reply:
Add a syntax variation
`Dim arr as Object()`

Reply:
`Dim arr() = New Object() { }` is not a ReDim operation but ArrayInitializer